Compiler error with c++ templates using teensy platform only

I can compile this code using gcc or the native platform in platformio, but I am getting an error when I build for the Teensy4.1:

```
template <typename T>
inline T saturate(T val, T lower, T upper)
{
   T val1 = val > lower ? val : lower;
   T val2 = val1 < upper ? val1 : upper;
   return val2;
}

int main()
{
    double v = 1.234;
    saturate(v, 0.0, 90.0);
    return 0;
}
```

The error I get is:
`src/exec/MainExec.cpp: In function 'int main()':
src/exec/MainExec.cpp:12:26: error: no matching function for call to 'saturate(double&, float, float)'
     saturate(v, 0.0, 90.0);`

Expected behavior: the compiler to make all three arguments doubles. 

Is this a bug with the teensy platform or with the compiler?

--- a/builder/frameworks/_bare_arm.py
+++ b/builder/frameworks/_bare_arm.py
@@ -34,8 +34,7 @@ env.Append(
         "-ffunction-sections",  # place each function in its own section
         "-fdata-sections",
         "-mthumb",
-        "-nostdlib",
-        "-fsingle-precision-constant"
+        "-nostdlib"
     ],
 
     CXXFLAGS=[
@@ -56,8 +55,7 @@ env.Append(
         "-Os",
         "-Wl,--gc-sections,--relax",
         "-mthumb",
-        "-Wl,--defsym=__rtc_localtime=$UNIX_TIME",
-        "-fsingle-precision-constant"
+        "-Wl,--defsym=__rtc_localtime=$UNIX_TIME"
     ],
 
     LIBS=["m", "stdc++"]
@@ -89,5 +87,19 @@ if "BOARD" in env:
         ],
         LINKFLAGS=[
             "-mcpu=%s" % env.BoardConfig().get("build.cpu")
+        ]
+    )
+
+if env.BoardConfig().get("build.core", "") != "teensy4":
+    env.Append(
+        ASFLAGS=[
+            "-mno-unaligned-access",
+        ],
+        CCFLAGS=[
+            "-mno-unaligned-access",
+            "-fsingle-precision-constant"
+        ],
+        LINKFLAGS=[
+            "-fsingle-precision-constant"
         ]
     )

--- a/builder/frameworks/arduino.py
+++ b/builder/frameworks/arduino.py
@@ -145,8 +145,7 @@ elif "BOARD" in env and BUILD_CORE in ("teensy3", "teensy4"):
             "-fdata-sections",
             "-mthumb",
             "-mcpu=%s" % env.BoardConfig().get("build.cpu"),
-            "-nostdlib",
-            "-fsingle-precision-constant"
+            "-nostdlib"
         ],
 
         CXXFLAGS=[

